common/src/test/java/tachyon/security/authentication/AuthenticationFactoryTest.java

```diff
@@ -22,49 +42,235 @@
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
 
-    AuthenticationFactory authenticationFactory = new AuthenticationFactory(tachyonConf);
+    AuthenticationFactory authenticationFactory = new AuthenticationFactory(mTachyonConf);
 
-    Assert.assertEquals(AuthenticationFactory.AuthType.SIMPLE.getAuthName(),
-        authenticationFactory.getAuthTypeStr());
+    Assert.assertEquals(AuthenticationFactory.AuthType.SIMPLE, authenticationFactory.getAuthType());
   }
 
   @Test
   public void getAuthTypeFromConfTest() {
-    TachyonConf tachyonConf = new TachyonConf();
     AuthenticationFactory.AuthType authType;
 
     // should return a SIMPLE AuthType with conf "SIMPLE"
-    tachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
-    authType = AuthenticationFactory.getAuthTypeFromConf(tachyonConf);
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
+    authType = AuthenticationFactory.getAuthTypeFromConf(mTachyonConf);
     Assert.assertEquals(AuthenticationFactory.AuthType.SIMPLE, authType);
 
     // should return a SIMPLE AuthType with conf "simple"
-    tachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "simple");
-    authType = AuthenticationFactory.getAuthTypeFromConf(tachyonConf);
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "simple");
+    authType = AuthenticationFactory.getAuthTypeFromConf(mTachyonConf);
     Assert.assertEquals(AuthenticationFactory.AuthType.SIMPLE, authType);
 
     // should throw exception with conf "wrong"
-    tachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "wrong");
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "wrong");
     mThrown.expect(IllegalArgumentException.class);
     mThrown.expectMessage("wrong is not a valid authentication type. Check the configuration "
         + "parameter " + Constants.TACHYON_SECURITY_AUTHENTICATION);
-    authType = AuthenticationFactory.getAuthTypeFromConf(tachyonConf);
+    authType = AuthenticationFactory.getAuthTypeFromConf(mTachyonConf);
+  }
+
+  /**
+   * In NOSASL mode, the TTransport used should be the same as Tachyon original code.
+   */
+  @Test
+  public void nosaslAuthenticationTest() throws Exception {
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "NOSASL");
+
+    // start server
+    startServerThread(mTachyonConf);
+
+    // create client and connect to server
+    TTransport client = createClient(null, null);
+    client.open();
+    Assert.assertTrue(client.isOpen());
+
+    // clean up
+    client.close();
+    mServer.stop();
+  }
+
+  /**
+   * In SIMPLE mode, the TTransport mechanism is PLAIN. When server authenticate the connected
+   * client user, it use {@link tachyon.security.authentication.SimpleAuthenticationProviderImpl}.
+   */
+  @Test
+  public void simpleAuthenticationTest() throws Exception {
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
```

Comment:
Nice catch! A null username should be check by client side when build `TSaslClientTransport`, and an exception is thrown.